server/src/controllers/orderController.ts

```diff
@@ -330,30 +331,32 @@ orderController.put('/refund/:id', async (req, res) => {
     if (!order) {
       return res.status(400).json({error: `Order ${orderID} does not exist`});
     }
-    if (!order.payment_intent) {
+    if (!order.payment_intent || order.order_status !== state.completed) {
```

Comment:
Just curious but is there ever a case where there isn't a payment_intent but the state of an order could be completed?

Reply:
I don't think so, I moved the everything over so `order_status` is the only indicator of the processing state. I left the  payment intent check there because it can possibly be null based on the db schema, but not our logic, so TS gives a possibly null error.